test/tests/serial/mail_smtponly.bats

```diff
@@ -1,64 +1,50 @@
-load "${REPOSITORY_ROOT}/test/test_helper/common"
+load "${REPOSITORY_ROOT}/test/helper/common"
+load "${REPOSITORY_ROOT}/test/helper/setup"
 
-function setup_file() {
-  docker run --rm -d --name mail_smtponly \
-    -v "$(pwd)/test/test-files":/tmp/docker-mailserver-test:ro \
-    -e SMTP_ONLY=1 \
-    -e PERMIT_DOCKER=network \
-    -e OVERRIDE_HOSTNAME=mail.my-domain.com \
-    -t "${NAME}"
+BATS_TEST_NAME_PREFIX='[SMTP-Only] '
+CONTAINER_NAME='dms-test_smtp_only'
 
-  wait_for_finished_setup_in_container mail_smtponly
-  wait_for_smtp_port_in_container mail_smtponly
-}
+function setup_file() {
+  _init_with_defaults
 
-function teardown_file() {
-  docker rm -f mail_smtponly
-}
+  local CUSTOM_SETUP_ARGUMENTS=(
+    --env SMTP_ONLY=1
+    --env PERMIT_DOCKER=network
+  )
 
-#
-# configuration checks
-#
+  _common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
 
-@test "checking configuration: hostname/domainname override" {
-  run docker exec mail_smtponly /bin/bash -c "cat /etc/mailname | grep my-domain.com"
-  assert_success
+  _wait_for_finished_setup_in_container "${CONTAINER_NAME}"
+  _wait_for_smtp_port_in_container
 }
 
-#
-# imap
-#
-
-@test "checking configuration: dovecot quota absent in postconf (disabled using SMTP_ONLY)" {
-  run docker exec mail_smtponly /bin/bash -c "postconf | grep 'check_policy_service inet:localhost:65265'"
-  assert_failure
-}
+function teardown_file() { _default_teardown ; }
 
-#
-# smtp
-#
-
-@test "checking smtp_only: mail send should work" {
-  run docker exec mail_smtponly /bin/sh -c "postconf smtp_host_lookup=no"
+@test "Dovecot quota absent in postconf" {
+  _run_in_container postconf
   assert_success
+  refute_output --partial "check_policy_service inet:localhost:65265'"
+}
 
-  _reload_postfix mail_smtponly
+# TODO: needs complete rework when proper DNS container is running for tests
+@test "sending mail should work" {
+  skip 'TODO: This test is absolutely broken and needs reworking!'
 
-  wait_for_smtp_port_in_container mail_smtponly
-  run docker exec mail_smtponly /bin/sh -c "nc 0.0.0.0 25 < /tmp/docker-mailserver-test/email-templates/smtp-only.txt"
+  # the value `no` is not even valid for `smtp_host_lookup`
+  # `smtp_host_lookup` seems to be deprecated too
+  _run_in_container postconf smtp_host_lookup=no
```

Comment:
[`smtp_host_lookup`](https://www.postfix.org/postconf.5.html#smtp_host_lookup) is not deprecated, but [`disable_dns_lookups`](https://www.postfix.org/postconf.5.html#disable_dns_lookups) is in favor of [`smtp_dns_support_level`](https://www.postfix.org/postconf.5.html#smtp_dns_support_level).

`smtp_host_lookup` is to configure how that lookup should be performed. If we want to opt-out we can disable with `smtp_dns_support_level=disabled`. That disables MX record lookups, but still defers regular hostname lookups to "native" as the Postfix docs describe it.

Chances are this isn't relevant at all to the test. I'll look through the history to see if there's any context, but it's doubtful.

---

<details>
<summary>References for revisiting this test in future</summary>

Based on the test name. The general approach in tests is to just send mail to the same DMS instance (_or sometimes a 2nd instance_), wait for the mail queue to empty and then check the mail directory it should have delivered to for a local account.

This test is only interested in if a mail was sent. `smtp-delivery.bats` has a bunch of test cases doing that, they look like this:

```bash
_run_in_container grep 'to=<user1~test@localhost.localdomain>, orig_to=<alias1~test@localhost.localdomain>' /var/log/mail/mail.log
assert_success
assert_output --partial 'status=sent'
_should_output_number_of_lines 1
```

And here's another from `mail_lmtp_ip.bats`:

```bash
@test "delivers mail to existing account" {
  _wait_for_smtp_port_in_container

  # Send a test mail:
  _run_in_container_bash "nc 0.0.0.0 25 < /tmp/docker-mailserver-test/email-templates/existing-user1.txt"
  assert_success

  # Verify delivery was successful, log line should look similar to:
  # postfix/lmtp[1274]: 0EA424ABE7D9: to=<user1@localhost.localdomain>, relay=127.0.0.1[127.0.0.1]:24, delay=0.13, delays=0.07/0.01/0.01/0.05, dsn=2.0.0, status=sent (250 2.0.0 <user1@localhost.localdomain> ixPpB+Zvv2P7BAAAUi6ngw Saved)
  local MATCH_LOG_LINE='postfix/lmtp.* status=sent .* Saved)'
  run timeout 60 docker exec "${CONTAINER_NAME}" bash -c "tail -F /var/log/mail/mail.log | grep --max-count 1 '${MATCH_LOG_LINE}'"
  assert_success
}
```

---

Since Dovecot is not available and no other mail storage is intended in this mode, a 2nd container to receive the mail probably makes more sense :sweat_smile: 

The `fail2ban.bats` test seems to handle this for SMTP auth by getting the container IP to use with `nc`, which would probably work?:

```bash
CONTAINER1_IP=$(_get_container_ip "${CONTAINER1_NAME}")
_run_in_container_explicit "${CONTAINER2_NAME}" bash -c "nc ${CONTAINER1_IP} 25 < /tmp/docker-mailserver-test/auth/smtp-auth-login-wrong.txt"
```

</details>

I'm fine with the `skip` you've added for now, and addressing this afterwards with a follow-up PR.

For now these lines at least don't seem to provide any value as you've documented (_thanks for pointing that out btw_ :grin: )

```suggestion
```

Reply:
Keeping this open for reference